clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2820,6 +2917,23 @@ bool Util::isSyclSpecConstantType(const QualType &Ty) {
   return matchQualifiedTypeName(Ty, Scopes);
 }
 
+bool Util::isPropertyListType(const QualType &Ty) {
+  return isSyclType(Ty, "property_list", true /*Tmpl*/);
+}
+
+bool Util::isSyclBufferLocationType(const QualType &Ty) {
+  const StringRef &Name = "buffer_location";
+  std::array<DeclContextDesc, 4> Scopes = {
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "cl"},
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "sycl"},
+      // TODO: this doesn't belong to property namespace, instead it shall be
+      // in its own namespace. Change it, when the actual implementation in SYCL
+      // headers is ready
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "property"},
```

Comment:
Extending this detecting approach is not really good, we already trying to avoid it https://github.com/intel/llvm/pull/2091 .

Reply:
I'm down with replacing this function when #2091 is ready.

Reply:
How #2091 will help you to do it?

Reply:
I think, that at least `property_list` can be added as enum argument of this attribute and here we query for it, or I'm missing something important?

Reply:
The idea is to make it generic, i.e. without arguments. See https://github.com/intel/llvm/pull/2091#discussion_r454241718
Although I'm not really sure how to make it work for specialization constants and any other classes that will need some handling that differs from handling for accessors.